xbmc/utils/StringUtils.h

```diff
@@ -92,6 +93,19 @@ class StringUtils
   static bool EndsWithNoCase(const std::string &str1, const char *s2);
 
   static std::string Join(const std::vector<std::string> &strings, const std::string& delimiter);
+  template <typename T, typename U>
+  static std::string Join(const std::map<T, U>& input_map, const std::string& delimiter = ",") {
+    if (input_map.empty())
+      return "";
+
+    std::ostringstream ss;
+    for (auto const& element : input_map)
+      ss << element.first << delimiter;
```

Comment:
Maybe use `std::to_string()` instead of string streams which are extremely slow?